internal/reconcile/rollback.go

```
@@ -0,0 +1,94 @@
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package reconcile
+
+import (
+	"context"
+	"fmt"
+
+	helmrelease "helm.sh/helm/v3/pkg/release"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	"github.com/fluxcd/pkg/runtime/conditions"
+	"github.com/fluxcd/pkg/runtime/logger"
+
+	helmv2 "github.com/fluxcd/helm-controller/api/v2beta2"
+	"github.com/fluxcd/helm-controller/internal/action"
+	"github.com/fluxcd/helm-controller/internal/release"
+	"github.com/fluxcd/helm-controller/internal/storage"
+)
+
+type Rollback struct {
+	configFactory *action.ConfigFactory
+}
+
+func (r *Rollback) Name() string {
+	return "rollback"
+}
+
+func (r *Rollback) Type() ReconcilerType {
+	return ReconcilerTypeRemediate
+}
+
+func (r *Rollback) Reconcile(ctx context.Context, req *Request) error {
+	var (
+		cur    = req.Object.Status.Current.DeepCopy()
+		logBuf = action.NewLogBuffer(action.LogV(ctrl.LoggerFrom(ctx), logger.InfoLevel), 10)
+	)
+
+	// Previous is required to determine what version to roll back to.
+	if req.Object.Status.Previous == nil {
+		return fmt.Errorf("%w: required to rollback", ErrNoPrevious)
+	}
+
+	// Run rollback action.
+	if err := action.Rollback(r.configFactory.Build(logBuf.Log, observeRollback(req.Object)), req.Object); err != nil {
+		// Mark failure on object.
+		req.Object.Status.Failures++
+		conditions.MarkFalse(req.Object, helmv2.RemediatedCondition, helmv2.RollbackFailedReason, err.Error())
+
+		// Return error if we did not store a release, as this does not
+		// affect state and the caller should e.g. retry.
+		if newCur := req.Object.Status.Current; newCur == nil || newCur == cur {
+			return err
+		}
+		return nil
```

could you explain why do we silently return here?